### ISSUE TITLE
syntax for jsonapi.version should be specified

### ISSUE THREAD

**Issue by freddrake** [json-api/json-api#1529]
The specification should specify how comparisons between values of the version member of the jsonapi object should be performed.

This will be especially important once there are more than a couple of versions of the specification around.

Adopting an existing specification for versioning seems better than inventing one; perhaps something like [Semantic Versioning](https://semver.org/).

**Comment by jelhan** [comment]
Good catch. I guess that's one of the cases on which one wrongly assumes it is clear - until 1.10` is released.

I don't think SemVer could be adopted. SemVer version consists of thee parts: `major.minor.patch` JSON:API versions consists of two parts only: `1.0` and `1.1`

Existing versions are changed in place _without_ bumping version number. This was used in the past for example if a change improves wording only. In SemVer this would be released as a patch.

I'm not sure if there is an existing versioning specification, which we can adopt taking our existing constraints into account.

For reference the version number is mentioned here in v1.1:

> The jsonapi object MAY contain a version member whose value is a string indicating the highest JSON API version supported.

Implicitly this requires to compare version numbers to determine, which is "the highest JSON API supported."
